2018-01-24-emacs-python/main.Rmd

```diff
@@ -0,0 +1,210 @@
+---
+title: "Usando scripts Python no Emacs"
+author: "Fernando Mayer"
+date: "24 de Janeiro de 2018"
+tags: ["python", "emacs"]
+categories: ["python", "emacs"]
+#output:
+#  html_fragment
+#-- Usar enquanto edita.
+output:
+  html_document:
+    theme: flatly
+---
+
+```{r, include = FALSE}
+# Carrega as configurações.
+source("../config/setup.R")
+```
+
+No início de Janeiro de 2018 iniciamos um grupo de estudos sobre o uso
+do Python para estatística. O objetivo era tentarmos descobrir como
+fazer em Python aquilo que fazemos rotineiramente em R. O resultado foi
+muito interessante e vimos que existe muita similaridade entre as duas
+linguagens. Abaixo está a divulgação que fizemos do grupo de estudos.
+
+<iframe src="https://www.facebook.com/plugins/post.php?href=https%3A%2F%2Fwww.facebook.com%2Fleg.ufpr%2Fposts%2F935006519988723&width=400&show_text=false&height=266&appId" width="400" height="266" style="border:none;overflow:hidden" scrolling="no" frameborder="0" allowTransparency="true"></iframe>
+
+Uma das primeiras coisas que analisamos, logo no início do curso, foram
+as IDEs (*Integrated Development Environment* ou Ambiente de
+Desenvolvimento Integrado) mais utilizadas para se escrever e executar
+códigos em Python. No mundo do R, sabemos que o onipresente
+[RStudio](https://www.rstudio.com) é o editor preferido da maioria dos
+usuários. No entanto, no mundo do Python não parece haver um único
+preferencial.
+
+Na categoria de uma IDE específica para a linguagem (como é o RStudio
+para o R), vimos primeiro o
+[Spyder](https://pythonhosted.org/spyder/index.html), por ser o editor
+distribuído junto com a plataforma [Anaconda](https://anaconda.org).
+Naturalmente, existe também o [Jupyter notebook](http://jupyter.org) que
+permite misturar código e texto em um único documento, mas por enquanto
+estávamos apenas interessados em editores para scripts "puros" de Python
+(arquivos `.py`). Fora o Spyder, sabemos que existem diversos bons
+editores de uso geral, que permitem a edição e possuem ferramentas
+comuns para várias linguagens. Entre estes editores estão o
+[Atom](https://atom.io) e o [Sublime](https://www.sublimetext.com), que
+tem suporte para a edição de scripts em Python.
+
+No entanto, como a ideia do grupo de estudos era se familiarizar com o
+Python, então nada melhor do que utilizarmos o mesmo editor que já
+utilizamos para códigos R. E esse editor é o
+[Emacs](https://www.gnu.org/software/emacs), utilizado por muitos aqui
+no LEG. O Emacs é um editor altamente customizável e de múltiplos
+propósitos. Seu alto índice de uso aqui no laboratório remonta aos
+tempos de uma era pré-RStudio, quando a edição de scripts R no Linux só
+era possível de duas formas: copiando código de um editor qualquer e
+colando no terminal ou usando o módulo [ESS](http://ess.r-project.org)
+(*Emacs Speaks Statistics*) para o Emacs.
```

Comment:
Eu incluiria que: a preferência pelo Emacs vai além dos recursos que ele dispõe para a linguagem R. O Emacs é usado para editar diversas outras linguagens, de programação a marcação, como LaTeX, Markdown, HTML, CSS, Shell, C, C++, JSON, etc, scripts do OpenBugs, do Stan, permite editar aquivos remotamente, além de inúmeras outras funcionalidades. O Emacs é um ambiente integrado para tudo. Eu arisco a dizer que a `geoR`, o livro da geoR, os pacotes R como `Rcitrus`, os arquivos de configuração dos servidores, os TCCs, dissertações e teses de todos no LEG foram feitos no Emacs.